### ISSUE TITLE
how does this work? how do i even set this up?

### ISSUE THREAD

**Issue by eklyst** [ugackMiner53/DiscordProximity#9]
this mod seems really good however i literally cannot see any setup instructions or guides anywhere. how do you connect it to discord? how does one even start the bot up? what do players even do once theyve installed it? so SO many questions and i dont even have a start to try figuring things out from because i have no idea how its even intended to work. pls just help me out and put like a brief idea of what you have to do to install it.

**Comment by ugackMiner53** [comment]
Hey, thanks for asking!
**TL;DR**
No setup, no discord server, nothing. Just install the mod on your computer and on your friend’s computer and join a Minecraft server together. That’s it.

**Why does that work?**
So, back in 2018, Discord wanted to sell games as a competitor to Steam, so they created the Discord Game Store, a place where you could buy games directly on Discord. Now, because they were acting as a Steam alternative, they needed to have the same features as Steam, and one of those features that Steam has is a voice chat service which developers can control. Discord ended up stealing the idea, and put it in their Discord Game SDK, a tool for developers to work with Discord to change a user’s activity status, among other things.

What this mod ends up doing is that every time you join a Minecraft server, it tells the Discord Game SDK to join you to an internal voice server (not a part of any server or anything that you could see). Then, anyone else who joins the same Minecraft server gets connected to the same Discord (internal) voice call. All the mod has to do then is fade people’s volume based on their distance!

**Comment by eklyst** [comment]
ok thanks
and what would i have to set up from the discord server side of things. do i need to link my disc? create a server? add a bot? input channelid somewhere?

**Comment by ugackMiner53** [comment]
Here's a comprehensive list of the things you have to do on Discord:
- **Absolutely Nothing!**

You don't even need a discord server! Here is the full list of steps that you have to do in order to get chatting with your friends:
1. Make sure that you have [Discord for Desktop](https://discord.com/download) installed and open.
2. Download [Fabric Launcher](https://fabricmc.net/)
3. Download [Fabric API](https://www.curseforge.com/minecraft/mc-mods/fabric-api)
4. Download [Discord Proximity](https://github.com/ugackMiner53/DiscordProximity/releases)
5. Install Fabric using the Fabric Launcher
6. Drag Fabric API and Discord Proximity into your `mods/` folder
7. Open Minecraft
8. Join a Minecraft Server

Repeat these steps on a friend's computer, and you should be able to hear them!
(Although do note, because of issue #1, both you and your friend need to type in the same server address in Minecraft, so `play.hypixel.net` and `mc.hypixel.net` won't work, but `play.hypixel.net` and `play.hypixel.net` _will_ work.

**Comment by ugackMiner53** [comment]
If you want to double check that the mod is working, you can run `/discordprox lobby list` in Minecraft and it should show you the Discord username of the first person to join the server. You could also run `/discordprox config`, and Discord should open up with a nice microphone config screen.

**Comment by DarianLStephens** [comment]
I was very surprised, too, but the mod literally _just works_ when it's installed.
Refer to my issue here: #5
